src/BloomExe/Edit/EditingModel.cs

```diff
@@ -671,13 +671,13 @@ public void HandleAddNewPageKeystroke(string unused)
 				if (CanDuplicatePage)
 				{
 					AddNewPageBasedOnTemplate(this._pageSelection.CurrentSelection.IdOfFirstAncestor);
-					return;
-				}
-				else
-				{
-					var idOfFirstPageInTemplateBook = CurrentBook.FindTemplateBook().GetPageByIndex(0).Id;
-					AddNewPageBasedOnTemplate(idOfFirstPageInTemplateBook);
+					if (_addNewSuccess)
```

Comment:
Just wrap your call to AddMessageEventListener:

_view.AddMessageEventListener("addPage", (id) => AddNewPageBasedOnTemplate(id));